src/scripts/action/sch/ActionAdloquium185.cpp

```diff
@@ -0,0 +1,83 @@
+{
+public:
+  ActionAdloquium185() :
+    ScriptAPI::ActionScript( 185 )
+  {
+  }
+
+  void onExecute( Sapphire::World::Action::Action& action ) override
+  {
+    auto pTarget = action.getHitChara();
+    if( pTarget )
+    {
+      // still pull out the lut entry to get potency values etc.
+      auto lutEntry = action.getActionEntry();
+
+      // do healing part
+      auto heal = action.calcHealing( lutEntry.healPotency );
+      heal.first = Math::CalcStats::applyHealingReceiveMultiplier( *pTarget, heal.first );
+      action.getEffectbuilder()->heal( pTarget, pTarget, heal.first, heal.second );
+
+      float shieldValue = heal.first * 1.25f;
+      // apply new galvanize when not existing or larger than existing one
+      auto oldEffect = pTarget->getStatusEffectById( STATUS_ID_GALVANIZE );
+      if( !oldEffect.second || oldEffect.second->getEffectEntry().effectValue1 <= shieldValue )
+      {
+        World::Action::StatusEffectEntry effectEntry;
+        effectEntry.effectType = static_cast< uint32_t >( Common::StatusEffectType::Shield );
+        effectEntry.effectValue1 = static_cast< int32_t >( shieldValue );
+        auto pNewEffect = action.createStatusEffect( STATUS_ID_GALVANIZE, action.getSourceChara(), pTarget, 30000, 3000 );
+        pNewEffect->replaceEffectEntry( effectEntry );
+        action.getEffectbuilder()->applyStatusEffect( pTarget, action.getSourceChara(), pNewEffect );
+      }
+      else
+        action.getEffectbuilder()->statusNoEffect( pTarget, STATUS_ID_GALVANIZE );
+
+      if( heal.second == Common::ActionHitSeverityType::CritHeal )
+      {
+        // apply catalyze when crit, same rule as galvanize
+        oldEffect = pTarget->getStatusEffectById( STATUS_ID_CATALYZE );
+        if( !oldEffect.second || oldEffect.second->getEffectEntry().effectValue1 <= shieldValue )
+        {
+          World::Action::StatusEffectEntry effectEntry;
+          effectEntry.effectType = static_cast< uint32_t >( Common::StatusEffectType::Shield );
+          effectEntry.effectValue1 = static_cast< int32_t >( shieldValue ); // same shield value
+          auto pNewEffect = action.createStatusEffect( STATUS_ID_CATALYZE, action.getSourceChara(), pTarget, 30000, 3000 );
+          pNewEffect->replaceEffectEntry( effectEntry );
+          action.getEffectbuilder()->applyStatusEffect( pTarget, action.getSourceChara(), pNewEffect );
+        }
+        else
+          action.getEffectbuilder()->statusNoEffect( pTarget, STATUS_ID_CATALYZE );
+      }
+    }
+  }
+
+  void onStart( Sapphire::World::Action::Action& action ) override
+  {
+    /*
+      don't run generic action handler for this action.
+      for simpler actions like cure 1 we only want to apply the freecure proc in script,
+      and let the generic handler do the heal so we don't have to copy heal code into scripts,
+      unless in cases like adlo, the healing result matters so we have to.
+    */
+    action.disableGenericHandler();
```

Comment:
it says clearly in the comment why we want to have both script and normal handling at the same time.

Reply:
I _don't_ really like this because it seems more like a hack than anything else and I can see some merit in doing it this way but I'd be inclined to not do it this way because the split responsibility makes it less obvious what's going on. But I think if you did the inverse and explicitly enabled it from a script you'd have the same problem anyway so I guess it doesn't matter much